lowstepper_wasm_demo/src/App.tsx

```diff
@@ -180,39 +216,323 @@ export default function App() {
+                }}>
+                </button>
+              </div>
+            </div>
+            <h3 className="parameterName">MORPH</h3>
+        </div>
+
+        <div className="trigWrapper">
+          <div>
+              <div className="buttonWrapper red">
+                <button className="button" onMouseDown={(e) =>  {
+                  setState({ ...state, "gate": true })
+                }}
+                onMouseUp={(e) => {
+                  setState({ ...state, "gate": false })
+                }}
+                onMouseOut={() => {
+                  setState({ ...state, "gate": false })
+                }}>
+                </button>
+              </div>
+            </div>
+            <h3 className="parameterName">CHUNKS</h3>
+        </div>
+
+        <div className="trigWrapper">
+          <div>
+              <div className="buttonWrapper blue">
+                <button className="button" onMouseDown={(e) =>  {
+                  setState({ ...state, "gate": true })
+                }}
+                onMouseUp={(e) => {
+                  setState({ ...state, "gate": false })
+                }}
+                onMouseOut={() => {
+                  setState({ ...state, "gate": false })
+                }}>
+                </button>
+              </div>
+            </div>
+        </div>
+
+        <div className="trigWrapper">
+          <div>
+              <div className="buttonWrapper blue">
+                <button className="button" onMouseDown={(e) =>  {
+                  setState({ ...state, "gate": true })
+                }}
+                onMouseUp={(e) => {
+                  setState({ ...state, "gate": false })
+                }}
+                onMouseOut={() => {
+                  setState({ ...state, "gate": false })
+                }}>
+                </button>
+              </div>
+            </div>
+        </div>
+
+        <div className="trigWrapper">
+          <div>
+              <div className="buttonWrapper blue">
+                <button className="button" onMouseDown={(e) =>  {
+                  setState({ ...state, "gate": true })
+                }}
+                onMouseUp={(e) => {
+                  setState({ ...state, "gate": false })
+                }}
+                onMouseOut={() => {
+                  setState({ ...state, "gate": false })
+                }}>
+                </button>
+              </div>
+            </div>
+        </div>
+
+      </div>
+      
+      <P5Wrapper sketch={sketch} {...state} />
+      <div className="container potentiometerWrapper">
         {knobs.map((currentKnob) => (
-          <div key={currentKnob.name}>
+          <div className="wrapper_potentiometer" key={currentKnob.name}>
+            <div className="bg_potentiometer"></div>
             <h2 className="parameterName"> {currentKnob.name}</h2>
+            <div className="bgKnob knobB">
+              <LimitedKnob
```

Comment:
add `onUpdate` to this component, then update corresponding props value per knob

Reply:
validate by seeing rate / morph / chunks change